A/AWS_MQTT/build_tarballs.jl

```diff
@@ -0,0 +1,139 @@
+cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
+	-DCMAKE_PREFIX_PATH=${prefix} \
+	-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+	-DCMAKE_BUILD_TYPE=Release \
+	-DBUILD_STATIC_LIBS=ON \
+	-DBUILD_SHARED_LIBS=OFF \
+	..
+cmake --build . -j${nproc}
+cmake --install .
+
+cd $WORKSPACE/srcdir/aws-c-common
+mkdir build && cd build
+cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
+	-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+	-DCMAKE_BUILD_TYPE=Release \
+	-DBUILD_STATIC_LIBS=ON \
+	-DBUILD_SHARED_LIBS=OFF \
+	..
+cmake --build . -j${nproc} --target install
+
+cd $WORKSPACE/srcdir/aws-c-cal
+mkdir build && cd build
+cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
+	-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+	-DCMAKE_PREFIX_PATH=${prefix} \
+	-DCMAKE_BUILD_TYPE=Release \
+	-DBUILD_STATIC_LIBS=ON \
+	-DBUILD_SHARED_LIBS=OFF \
+	..
+cmake --build . -j${nproc} --target install
+
+cd $WORKSPACE/srcdir/aws-c-io
+mkdir build && cd build
+cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
+    -DCMAKE_PREFIX_PATH=${prefix} \
+	-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+	-DCMAKE_BUILD_TYPE=Release \
+	-DBUILD_STATIC_LIBS=ON \
+	-DBUILD_SHARED_LIBS=OFF \
+	..
+cmake --build . -j${nproc} --target install
+
+cd $WORKSPACE/srcdir/aws-c-compression
+mkdir build && cd build
+cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
+    -DCMAKE_PREFIX_PATH=${prefix} \
+	-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+	-DCMAKE_BUILD_TYPE=Release \
+	-DBUILD_STATIC_LIBS=ON \
+	-DBUILD_SHARED_LIBS=OFF \
+	..
+cmake --build . -j${nproc} --target install
+
+cd $WORKSPACE/srcdir/aws-c-http
+mkdir build && cd build
+cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
+    -DCMAKE_PREFIX_PATH=${prefix} \
+	-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+	-DCMAKE_BUILD_TYPE=Release \
+	-DBUILD_STATIC_LIBS=ON \
+	-DBUILD_SHARED_LIBS=OFF \
+	..
+cmake --build . -j${nproc} --target install
+
+cd $WORKSPACE/srcdir/aws-c-mqtt
+mkdir build && cd build
+cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
+    -DCMAKE_PREFIX_PATH=${prefix} \
+	-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
+	-DCMAKE_BUILD_TYPE=Release \
+	-DBUILD_STATIC_LIBS=ON \
+	-DBUILD_SHARED_LIBS=OFF \
+	..
+cmake --build . -j${nproc} --target install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
+
+# The products that we will ensure are always built
+products = [
+    FileProduct(["lib/libaws-c-mqtt.a", "lib64/libaws-c-mqtt.a"], :libawsmqtt),
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    # TODO: this is needed only for Windows, but it looks like filtering
+    # platforms for `HostBuildDependency` is broken
+    HostBuildDependency("NASM_jll")
+]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6", preferred_gcc_version = v"10.2.0")
```

Comment:
Isn't there an older version of GCC that can build this?  10 is relatively recent, we prefer older versions for maximum compatibility.

Reply:
I can try older versions, but these AWS libs explicitly only support toolchains up to 5 years old.